src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/TextDifferencing/TextDiffer.IntArray.cs

```diff
@@ -118,7 +118,7 @@ private static void ReturnArray(int[] array, bool clearArray = false)
                 // Does this index fall within page? If not, acquire the appropriate page.
                 if (index < page.Start || index >= page.Start + page.Length)
                 {
-                    page = _pages[index % PageSize];
+                    page = _pages[index / PageSize];
```

Comment:
This logic only makes sense with division. Also this error does not happen with small files because of the `if`.